### PR TITLE
Update the appearance of the template details popover

### LINKED ISSUE #31076: It should be possible to delete new custom block  templates, and revert customised block templates in the template editor
In the post editor it is now possible to:

* Create a new custom block template on the fly
* If a block template is used, edit it visually in the template editor

Screenshot 2021-04-22 at 10 13 29

In the case of user-generated block templates, the UI should provide a way for users to delete that template.

In the case of customised block theme templates it should be possible to revert those customisations and re-instate the theme default.

The Site Editor already offers a way to revert customised block theme templates:

Screenshot 2021-04-22 at 10 18 34

This is something we can consider bringing to the template editor (albeit without the "Browse all templates" button for now). The same design can be utilised for deletion. Here's a rough mockup:

Screenshot 2021-04-22 at 10 27 20

Since this would be a destructive action, we might want to expose a confirmation dialog that briefly explains the consequences of this action, something like:

> Delete "My Custom Template"?
> All posts that use this template will be affected.
> [Cancel] [Delete]

It we could provide a count of how many posts would be affected, that would be even better :) 

- [ ] Provide a way to revert customised theme-supplied block templates in the template edior
- [ ] Provide a way to delete custom block templates in the template editor

### PR DESCRIPTION
**Pull request by jameskoster**
Here's the before:

Screenshot 2021-05-20 at 12 23 16

Here's the after:

Screenshot 2021-05-20 at 12 58 08

Changes:

* Narrowed the width a bit
* Added helper text for the title input
* Added a separator between the title and delete controls
* Increased the weight of the delete button

### CODE REVIEW

File: packages/edit-post/src/components/header/template-title/edit-template-title.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ export default function EditTemplateTitle() {
 		<TextControl
 			label={ __( 'Title' ) }
 			value={ templateTitle }
+			help={ __(
+				'Give the template a title that indicates its purpose, e.g. "Full Width"'
```
Comment (jasmussen) [inline packages/edit-post/src/components/header/template-title/edit-template-title.js]: Period at the end?

File: packages/edit-post/src/components/header/template-title/style.scss [hunk 2]
```diff
@@ -13,5 +13,22 @@
...
-	min-width: 360px;
+	min-width: 280px;
+}
+
+.edit-post-template-top-area__second-menu-group {
+	margin-left: - $grid-unit-15;
+	margin-right: - $grid-unit-15;
+	padding: $grid-unit-15;
+	padding-bottom: 0;
+	border-top: 1px solid $gray-300;
```
Comment (jasmussen) [inline packages/edit-post/src/components/header/template-title/style.scss]: Nitpick, can be $border-width

File: packages/edit-post/src/components/header/template-title/style.scss [hunk 3]
```diff
@@ -13,5 +13,22 @@
 }
 
 .edit-post-template-top-area__popover .components-popover__content {
-	min-width: 360px;
+	min-width: 280px;
+}
+
+.edit-post-template-top-area__second-menu-group {
+	margin-left: - $grid-unit-15;
```
Comment (jasmussen) [inline packages/edit-post/src/components/header/template-title/style.scss]: Oh interesting with the space after the minus! I'd usually do -$grid-unit-15 for negative numbers. Not really something that needs fixing, just stood out.

Comment (jameskoster) [reply packages/edit-post/src/components/header/template-title/style.scss]: I've had issues with not putting a space in the past, so put one in habitually now. I'll check if it's actually needed though.

Comment (jasmussen) [reply packages/edit-post/src/components/header/template-title/style.scss]: I assume it's fine, just looking at the code here while I'm rebooting my environment testing your other PR.

**Review by jasmussen — APPROVED** [review]
Before:

before

After:

after

The branch needs a rebase, and the focus style of the dropdown could use a little love. That last part doesn't have to be here though as it's also an issue in trunk. But this is a nice improvement, thanks!

**Comment by iamtakashi** [comment]
A quick question. I no longer see the delete template option. Is there a new way to delete it now? Thanks in advance!

**Comment by jameskoster** [comment]
Interesting, neither do I. I don't recall it intentionally being removed so you might have uncovered a bug :)

For now you can delete custom templates in the Template list:
Screenshot 2022-12-01 at 16 48 07

**Comment by iamtakashi** [comment]
That'd make sense, but the thing is, I don't see the three-dots menu either. That feels familiar, but I haven't seen it for a while. For example, I don't see them on the site TT2 (or TT3) is activated. What am I missing?

**Comment by iamtakashi** [comment]
Adding more context. I realised that I had a template I didn't need. I deleted the template file from the theme directory. But the template is still listed in the list in the editor even though there isn't the template file in the theme anymore. I tried to delete the cache, but it didn't help, so I wanted to know how to delete it from the list.

**Comment by jameskoster** [comment]
> That'd make sense, but the thing is, I don't see the three-dots menu either

You can only delete templates that were added in-product. If the template exists in the file system (IE supplied by the theme) then it cannot be deleted.

> I deleted the template file from the theme directory. But the template is still listed in the list in the editor even though there isn't the template file in the theme anymore.

Do you remember if you customised the template? A customised template will remain, even if you delete the template file in the theme. I _think_ this might be a bug... customised templates should be revertable (resets them back to the file supplied by the theme), but if the file is deleted from the file system then there's nothing to revert to. In this case the delete action should appear. It's a bit of an edge case, but worth opening an issue if you have the time.

**Comment by iamtakashi** [comment]
Thank you for the detailed reply. Much appreciated.

> You can only delete templates that were added in-product. If the template exists in the file system (IE supplied by the theme) then it cannot be deleted.

I see. That makes sense for the majority of the users to stop accidentally deleting the templates. 

> Do you remember if you customised the template? A customised template will remain, even if you delete the template file in the theme.

Unless I'm missing something obvious, in my experience, even non-customised templates that have been deleted from the theme folder keep showing up on the template list... and if you open the template, you'll see this... With this in mind, it feels reasonable to think the template should be gone from the list when it's removed from a theme.

iamtakashi jurassic tube_wp-admin_site-editor php_postType=wp_template postId=bitacora%2F%2Farchive (1)

I tried clearing the cache and deactivating/activating the theme, but only rebooting the dev environment removed the unwanted template from the list.

If the template was customised, it remains on the list, and I can see the customised design when I open it.

Thanks again, @jameskoster, for sharing your knowledge. It feels like it's worth creating an issue/question about it.

**Comment by jameskoster** [comment]
> even non-customised templates that have been deleted from the theme folder keep showing up on the template list...

Yeah I can't explain that, so I agree it would appear to be a bug. Definitely worth opening an issue if you have the time.

**Pull request merged**